src/components/TradeAssetSearch/hooks/useGetCustomTokenPriceQuery.tsx

```diff
@@ -0,0 +1,80 @@
+import { type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import { useQueries, type UseQueryResult } from '@tanstack/react-query'
+import axios from 'axios'
+import { mergeQueryOutputs } from 'react-queries/helpers'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+
+type UseGetCustomTokenPricesQueryProps = {
+  assetIds: AssetId[]
+}
+
+type UseGetCustomTokenPricesQueryReturn = {
+  assetId: AssetId
+  marketData: ZerionMarketData | undefined
+}[]
+
+const ZERION_BASE_URL = 'https://zerion.shapeshift.com/'
+
+const axiosInstance = axios.create()
+
+type ZerionMarketData = {
+  price: number
+  total_supply: number
+  circulating_supply: number
+  fully_diluted_valuation: number
+  market_cap: number
+  changes: {
+    percent_1d: number
+    percent_30d: number
+    percent_90d: number
+    pecent_365d: number
+  }
+}
+
+export const getTokenMarketData = async (
+  assetId: AssetId,
+): Promise<ZerionMarketData | undefined> => {
+  const options = {
+    method: 'GET' as const,
+    baseURL: ZERION_BASE_URL,
+  }
+
+  const { assetReference } = fromAssetId(assetId)
+  const url = `/fungibles/${assetReference}`
+  const payload = { ...options, url }
+  const { data: res } = await axiosInstance.request(payload)
+
+  return res.data.attributes.market_data
```

Comment:
don't really need the instance and options for what we are doing here if we want less lines :)
```suggestion
  const { assetReference } = fromAssetId(assetId)
  const url = `${ZERION_BASE_URL}/fungibles/${assetReference}`
  const { data: res } = await axios.get<ZerionFungibles>(url)
  return res.data.attributes.market_data
```

Reply:
https://github.com/shapeshift/web/pull/7370/commits/1c8f6da041d9491392285d0b470517a76cce306d